File: back/DevC/src/main/java/com/back/devc/global/response/ErrorResponse.java
package com.back.devc.global.response;

import com.back.devc.global.exception.ErrorCodeSpec;

import java.time.LocalDateTime;
import java.util.Map;

public record ErrorResponse(
        String code,
        String message,
        LocalDateTime timestamp,
        Map<String, String> validation
) {
    public static ErrorResponse of(ErrorCodeSpec errorCode) {
        return new ErrorResponse(
                errorCode.getCode(),
                errorCode.getMessage(),
                LocalDateTime.now(),
                Map.of()
        );
    }

    public static ErrorResponse of(ErrorCodeSpec errorCode, Map<String, String> validation) {
        return new ErrorResponse(
                errorCode.getCode(),
                errorCode.getMessage(),
                LocalDateTime.now(),
                validation
        );
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/global/response/ErrorResponse.kt
package com.back.devc.global.response

import com.back.devc.global.exception.ErrorCodeSpec
import java.time.LocalDateTime

data class ErrorResponse(
    val code: String,
    val message: String,
    val timestamp: LocalDateTime,
    val validation: Map<String, String>
) {

    companion object {
        @JvmStatic
        fun of(errorCode: ErrorCodeSpec): ErrorResponse {
            return ErrorResponse(
                code = errorCode.code,
                message = errorCode.message,
                timestamp = LocalDateTime.now(),
                validation = emptyMap()
            )
        }

        @JvmStatic
        fun of(
            errorCode: ErrorCodeSpec,
            validation: Map<String, String>
        ): ErrorResponse {
            return ErrorResponse(
                code = errorCode.code,
                message = errorCode.message,
                timestamp = LocalDateTime.now(),
                validation = validation
            )
        }
    }
}